tests/test_cases/test_sim_standalone/test_sim_standalone.py

```diff
@@ -0,0 +1,194 @@
+###############################################################################
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+###############################################################################
+
+from unittest.case import TestCase
+import cocotb
+import logging
+from cocotb.log import SimLog
+from cocotb.triggers import Event, Timer
+from cocotb.scheduler import Scheduler
+from cocotb.utils import get_sim_time
+
+class TestSimStandalone(TestCase):
```

Comment:
I am assuming  `TestSimStandalone` class is a common class that is provided by this PR and the cocotb user will only be instantiating it. Am I correct? Or does every user need to rewrite this class as per his usecase?

Reply:
TestSimStandalone is just the unit test class. Each use-case would be expected to provide an implementation of the APIs in the class SimulatorStub below. The API implemented by the class SimulatorStub (eg register_timed_callback) is currently implemented by the C/C++ Python module that integrates with the simulator. 
My understanding of the consensus was to document the APIs implemented by that module (eg register_timed_callback), but not provide a class implementing those APIs as part of cocotb.

Reply:
> My understanding of the consensus was to document the APIs implemented by that module (eg register_timed_callback), but not provide a class implementing those APIs as part of cocotb.

I'd perhaps relax this to "not provide a class implementing those APIs _except as controlling an attached simulator_". I'm not opposed to swapping out the way we use the `simulator` module for a `Simulator` class of some kind.

Reply:
Some detail on that in https://github.com/cocotb/cocotb/pull/1987#issuecomment-659945859.